packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -238,6 +225,7 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
                 .customizableProperties(customizableProperties)
                 .modifiers(modifiers)
                 .state(state)
+                .pinned(modifiers.contains(ViewModifier.Pinned))
```

Comment:
Why do you have both on one side `modifiers` and `pinned`? either you rely on `modifiers` and add the pinned property in `ViewModifier` or you have a new attribute. Given the fact that you have a dedicated attribute `pinned` in the GraphQL schema and that in the GraphQL schema, the type `ViewModifier` does not show `pinned` as a potential value, I don't think that mixing the two would be a good idea.

```
enum ViewModifier {
  Normal
  Faded
  Hidden
}
```

And you can't really add `pinned` here in the GraphQL schema because if you do that, then `pinned` becomes a valid value for the "show / hide / fade" feature. So please remove the addition of `pinned` in the `ViewModifier.java` enum.

Reply:
Done